### ISSUE TITLE
Document full configs for secondary option examples

### ISSUE THREAD

**Issue by IAMtheIAM** [stylelint/stylelint#4808]
> Clearly describe the bug

The `selector-pseudo-element-no-unknown` rule needs to accept `true` and `false` so it can be overridden when extending from a predefined config which has it set to `true`.

> Which rule, if any, is the bug related to?

`selector-pseudo-element-no-unknown`

> What code is needed to reproduce the bug?

No code

> What stylelint configuration is needed to reproduce the bug?

        "selector-pseudo-element-no-unknown": false,

> Which version of stylelint are you using?

        "stylelint": "13.5.0",

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

 "Yes, it's related to SCSS nested properties."

> What did you expect to happen?

"No error."

> What actually happened (e.g. what warnings or errors did you get)?

```
Results in error: Invalid Option: Unexpected option value "false" for rule "selector-pseudo-element-no-unknown"
```

---

Rules in progress:

- [x] #8552 
- [ ] `annotation-no-unknown`
- [ ] `at-rule-allowed-list`
- [ ] `at-rule-descriptor-no-unknown`
- [ ] `at-rule-descriptor-value-no-unknown`
- [ ] `at-rule-disallowed-list`
- [ ] `at-rule-empty-line-before`
- [ ] `at-rule-no-deprecated`
- [ ] `at-rule-no-unknown`
- [ ] `at-rule-no-vendor-prefix`
- [ ] `at-rule-prelude-no-invalid`
- [ ] `at-rule-property-required-list`
- [ ] `block-no-empty`
- [ ] `color-function-alias-notation`
- [ ] `color-function-notation`
- [ ] `color-hex-alpha`
- [ ] `color-hex-length`
- [ ] `color-named`
- [ ] `color-no-hex`
- [ ] `color-no-invalid-hex`
- [ ] `comment-empty-line-before`
- [ ] `comment-no-empty`
- [ ] `comment-pattern`
- [ ] `comment-whitespace-inside`
- [ ] `comment-word-disallowed-list`
- [ ] `container-name-pattern`
- [ ] `custom-media-pattern`
- [ ] `custom-property-empty-line-before`
- [ ] `custom-property-no-missing-var-function`
- [ ] `custom-property-pattern`
- [ ] `declaration-block-no-duplicate-custom-properties`
- [ ] `declaration-block-no-duplicate-properties`
- [ ] `declaration-block-no-redundant-longhand-properties`
- [ ] `declaration-block-no-shorthand-property-overrides`
- [ ] `declaration-block-single-line-max-declarations`
- [ ] `declaration-empty-line-before`
- [ ] `declaration-no-important`
- [ ] `declaration-property-max-values`
- [ ] #8553 
- [ ] `declaration-property-unit-disallowed-list`
- [ ] `declaration-property-value-allowed-list`
- [ ] `declaration-property-value-disallowed-list`
- [ ] `declaration-property-value-keyword-no-deprecated`
- [ ] `declaration-property-value-no-unknown`
- [ ] `font-family-name-quotes`
- [ ] `font-family-no-duplicate-names`
- [ ] `font-family-no-missing-generic-family-keyword`
- [ ] `font-weight-notation`
- [ ] `function-allowed-list`
- [ ] `function-calc-no-unspaced-operator`
- [ ] `function-disallowed-list`
- [ ] `function-linear-gradient-no-nonstandard-direction`
- [ ] `function-name-case`
- [ ] `function-no-unknown`
- [ ] `function-url-no-scheme-relative`
- [ ] `function-url-quotes`
- [ ] `function-url-scheme-allowed-list`
- [ ] `function-url-scheme-disallowed-list`
- [ ] `hue-degree-notation`
- [ ] `import-notation`
- [ ] `keyframe-block-no-duplicate-selectors`
- [ ] `keyframe-declaration-no-important`
- [ ] `keyframe-selector-notation`
- [ ] `keyframes-name-pattern`
- [ ] `layer-name-pattern`
- [ ] `length-zero-no-unit`
- [ ] `lightness-notation`
- [ ] `max-nesting-depth`
- [ ] `media-feature-name-allowed-list`
- [ ] `media-feature-name-disallowed-list`
- [ ] `media-feature-name-no-unknown`
- [ ] `media-feature-name-no-vendor-prefix`
- [ ] `media-feature-name-unit-allowed-list`
- [ ] `media-feature-name-value-allowed-list`
- [ ] `media-feature-name-value-no-unknown`
- [ ] `media-feature-range-notation`
- [ ] `media-query-no-invalid`
- [ ] `named-grid-areas-no-invalid`
- [ ] `no-descending-specificity`
- [ ] `no-duplicate-at-import-rules`
- [ ] `no-duplicate-selectors`
- [ ] `no-empty-source`
- [ ] `no-invalid-double-slash-comments`
- [ ] `no-invalid-position-at-import-rule`
- [ ] `no-irregular-whitespace`
- [ ] `no-unknown-animations`
- [ ] `no-unknown-custom-media`
- [ ] `no-unknown-custom-properties`
- [ ] `number-max-precision`
- [ ] `property-allowed-list`
- [ ] `property-disallowed-list`
- [ ] `property-no-unknown`
- [ ] `property-no-vendor-prefix`
- [ ] `rule-empty-line-before`
- [ ] `rule-selector-property-disallowed-list`
- [ ] `selector-anb-no-unmatchable`
- [ ] `selector-attribute-name-disallowed-list`
- [ ] `selector-attribute-operator-allowed-list`
- [ ] `selector-attribute-operator-disallowed-list`
- [ ] `selector-attribute-quotes`
- [ ] `selector-class-pattern`
- [ ] `selector-combinator-allowed-list`
- [ ] `selector-combinator-disallowed-list`
- [ ] `selector-disallowed-list`
- [ ] `selector-id-pattern`
- [ ] `selector-max-attribute`
- [ ] `selector-max-class`
- [ ] `selector-max-combinators`
- [ ] `selector-max-compound-selectors`
- [ ] `selector-max-id`
- [ ] `selector-max-pseudo-class`
- [ ] `selector-max-specificity`
- [ ] `selector-max-type`
- [ ] `selector-max-universal`
- [ ] `selector-nested-pattern`
- [ ] `selector-no-qualifying-type`
- [ ] `selector-no-vendor-prefix`
- [ ] `selector-not-notation`
- [ ] `selector-pseudo-class-allowed-list`
- [ ] `selector-pseudo-class-disallowed-list`
- [ ] `selector-pseudo-class-no-unknown`
- [ ] `selector-pseudo-element-allowed-list`
- [ ] `selector-pseudo-element-colon-notation`
- [ ] `selector-pseudo-element-disallowed-list`
- [ ] `selector-pseudo-element-no-unknown`
- [ ] `selector-type-case`
- [ ] `selector-type-no-unknown`
- [ ] `shorthand-property-no-redundant-values`
- [ ] `string-no-newline`
- [ ] `syntax-string-no-invalid`
- [ ] `time-min-milliseconds`
- [ ] `unit-allowed-list`
- [ ] `unit-disallowed-list`
- [ ] `unit-no-unknown`
- [ ] `value-keyword-case`
- [ ] `value-no-vendor-prefix`

**Comment by IAMtheIAM** [comment]
I found that I can do this:

```
        "selector-pseudo-element-no-unknown": [true, {
            "ignorePseudoElements": ["ng-deep"]
        }],

```
But I did not understand this clearly from the documentation. It would be really helpful if the rules showed the full config in the options and secondary options, just an idea.

**Comment by hudochenkov** [comment]
> The selector-pseudo-element-no-unknown rule needs to accept true and false so it can be overridden when extending from a predefined config which has it set to true.

Use `null` to [turn off a rule](https://stylelint.io/user-guide/configure#rules):

```js
"selector-pseudo-element-no-unknown": null
```

> But I did not understand this clearly from the documentation. It would be really helpful if the rules showed the full config in the options and secondary options, just an idea.

I would agree with you. Style of configuration examples in rule description are rather not user-friendly, and require quite some knowledge about how stylelint works and how it's configured.

E. g. this is what we have in [`selector-pseudo-element-no-unknown`](https://stylelint.io/user-guide/rules/selector-pseudo-element-no-unknown#ignorepseudoelements-regex-string):

> Given:
>
> ```js
> ["/^my-/", "pseudo-element"]
> ```

This would be much more helpful for users:

> Given:
>
> ```js
> [true, { "ignorePseudoElements": ["/^my-/", "pseudo-element"] }]
> ```

**Comment by jeddy3** [comment]
> This would be much more helpful for users:

Agreed. I'll label the issue as ready to go.

@IAMtheIAM Please consider [contributing](https://stylelint.io/contributing) these changes if you have time.

**Comment by Mouvedia** [comment]
The initial issue is covered by #5091.

**Comment by ybiquitous** [comment]
I agree with @Mouvedia.

If PR #6250 is merged, users will receive a more helpful message, and we've already provided the [doc for turning off a rule](https://stylelint.io/user-guide/configure/#rules). I think these are sufficient.

**Comment by ybiquitous** [comment]
I've closed PR #6250, so I've also completed this issue. Feel free to reopen it if anyone has objections.

**Comment by hudochenkov** [comment]
@ybiquitous I think this issue is not about disabling rules, but documentation of configuration in general. 

ESLint for every example [shows](https://eslint.org/docs/latest/rules/multiline-comment-style) full config value for a rule. Stylelint docs show only part of a value. See my comment above for an example. https://github.com/stylelint/stylelint/issues/4808#issuecomment-634998634

**Comment by ybiquitous** [comment]
Ah, I misunderstood it. Thanks for letting me know, @hudochenkov!

**Comment by Mouvedia** [comment]
> > Given:
> > ```js
> > ["/^my-/", "pseudo-element"]
> > ```
> 
> This would be much more helpful for users:
> 
> > Given:
> > ```js
> > [true, { "ignorePseudoElements": ["/^my-/", "pseudo-element"] }]
> > ```

That would require to edit many more markdown file other than this one.
e.g. alpha-value-notation, selector-pseudo-class-no-unknown, at-rule-no-unknown, selector-type-no-unknown, media-feature-name-no-unknown, annotation-no-unknown, etc.

I don't think this is necessary or the priority should be very low.

**Comment by ybiquitous** [comment]
I decided to work on this issue, so I've added a rule list in this issue's description to help us understand the progress. Welcome any help!
